nupic/research/frameworks/sigopt/mixins/sgd_params.py

```diff
@@ -0,0 +1,70 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import math
+
+from torch.optim import SGD
+
+
+class SGDParams(object):
+
+    def update_config_with_suggestion(self, config, suggestion):
+        """
+        Given a SigOpt suggestion, update the optimizer_args with SGD optimizer params.
```

Comment:
Nothing in this class seems specific to SGD. Other optimizers could have learning rate, weight_decay, etc. If it is specific to SGD, it should check that the optimizer is indeed SGD.

Reply:
I thought it would be helpful to be explicit about what each mixin is intended to do. If one wants to optimize Adam parameters, then it's fairly straightforward to introduce a new mixin. I'll add a check for the class type.